syntax = "proto3";

import "google/protobuf/empty.proto";

service Lvt {
  rpc CreateVehicle (VehicleLocation) returns (VehicleId) {}

  rpc GetVehicle (VehicleId) returns (VehicleValues) {}

  rpc UpdateVehicle (VehicleValues) returns (google.protobuf.Empty) {}

  rpc DeleteVehicle (VehicleId) returns (google.protobuf.Empty) {}

  rpc GetVehiclesInArea (Area) returns (stream VehicleValues) {}
}

message VehicleLocation {
  double latitude = 1;
  double longitude = 2;
}

message VehicleId {
  string id = 1;
}

message VehicleValues {
  string id = 1;
  double latitude = 2;
  double longitude = 3;
}

message Area {
  double fromLatitude = 1;
  double toLatitude = 2;
  double fromLongitude = 3;
  double toLongitude = 4;
}